app/src/main/java/com/imdb_compose/MainActivity.kt

```diff
@@ -1,47 +1,1195 @@
+                                modifier = Modifier.padding(start = 8.dp),
+                                onClick = { navController.navigate(Navigator.CategoryPage(catagory = catagory)) }
+                            ) {
+                                Text(
+                                    text = "See all",
+                                    modifier = Modifier.padding(start = 8.dp),
+                                    color = blue400,
+                                    fontSize = MaterialTheme.typography.headlineSmall.fontSize,
+                                    fontStyle = MaterialTheme.typography.headlineSmall.fontStyle,
+                                    fontWeight = FontWeight.Bold
+                                )
+                            }
+                        }
+                    }
+
+                    Spacer(modifier = Modifier.height(8.dp))
+
+                    when(catagory) {
+                        "Tv airing today" -> TvBox(catagory, viewModel.airingTodayTv.collectAsState(), navController)
+                        "Trending tv" -> TvBox(catagory, viewModel.trendingTv.collectAsState(), navController)
+                        "Top box office" -> BoxOfficeBox(catagory, viewModel.boxOffice, navController)
+                        "Upcoming movies" -> UpcommingBox(catagory, viewModel.upcomingMovies.collectAsState(), viewModel, navController)
+                        "Popular actors" -> PersonBox(catagory, viewModel.popularPersons.collectAsState(), navController)
+                        "Trending people" -> PersonBox(catagory, viewModel.trendingPersons.collectAsState(), navController)
+                        "Movies of the week" -> MovieBox(catagory, viewModel.movieListOfWeek.collectAsState(), viewModel, navController)
+                        "Trending movies" -> MovieBox(catagory, viewModel.trendingMovies.collectAsState(), viewModel, navController)
+                        else -> MovieBox(catagory, viewModel.noMovies.collectAsState(), viewModel, navController)
+                    }
+                }
+            }
+        }
+        Spacer(modifier = Modifier.height(16.dp))
+    }
+}
+
+@Composable
+fun MovieBox(
+    catagory: String,
+    movies: State<MovieList?>,
+    viewModel: HomeScreenViewModel,
+    navController: NavController
+) {
+    Box(modifier = Modifier.padding(start = 8.dp)) {
+        LazyRow {
+            movies.value?.results?.filter {
+                it.poster_path != null
+            }?.forEachIndexed { i, movie ->
+                item {
+                    PosterBoxA(
+                        catagory = catagory,
+                        movie = movie,
+                        show = null,
+                        navController = navController,
+                        rank = "${ i + 1 }"
+                    )
+                    Spacer(modifier = Modifier.width(8.dp))
+                }
+            }
+        }
+    }
+}
+
+@Composable
+fun TvBox(
```

Comment:
`TvBox` and `MovieBox` are almost exactly the same. This tells me you're passing in too much information. At this level, you shouldn't care if you have a TV show or a movie; you just need some text to render on the screen.

You should only pass the data you need to render the box. You probably did it this way since you pass a list of either `TvList` or `MovieList` and to make this function generic you would need to create a third data type to pass only the info you needed. To avoid this, you should create two different Composable functions, one at the row level and one at the item level. Something like this:

```kotlin
@Composable
fun MyLazyRow(modifier: Modifier = Modifier, content: LazyListScope.() -> Unit) {
     Box(modifier = modifier.padding(start = 8.dp)) {
         LazyRow(content = content)
      }
}

@Composable
fun PosterBox(category: String, title: String, rank: String, onClick: () -> Unit) { ... }

@Composable
fun MyTvUse(tvShows: State<TvList?>) {
   MyLazyRow {
       tvShows.value?.results?.filter { ... }?.forEachIndexed { i, show ->
           item {
               PosterBox(...)
           }
       }
   }
}
```

This way, you can reuse `PosterBox` and `MyLazyRow` for both TV shows and Movies and possibly for people as well.

I'm still not a fan of that chain of calls to get the TV shows but we can tackle that in the next iteration.